Add all supported WP versions to Travis CI check

peterfabian opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Based on p7bje6-1Xe-p2, adding test matrix for all supported WP versions.

### How to test the changes in this Pull Request:

1. Check that tests for WP 5.0 - WP 5.4 run on Travis and are green.
2. Done!

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

peterfabian committed:
Minimum version is no longer PHP 5.6, but PHP 7.0

peterfabian committed:
Test all supported WordPress versions.

claudiosanches reviewed (changes requested):
Tests are failing, also it takes a long time to finish, so I think we need to evaluate if it will impact in our processing, since any change may takes 30 minutes or more to get tested on Travis.

peterfabian committed:
Reduce the number of tests to run.

Just test with one PHP version, so that it doesn't take more than 30 minutes to finish.

claudiosanches reviewed (approved):
Now looks great!
I also solved the broke unit tests in https://github.com/woocommerce/woocommerce/pull/26195

ObliviousHarmony committed:
Merge branch 'master' into fix/supported-wp-version-travis-update
